pkg/provider/kubernetes/gateway/tlsroute.go

```diff
@@ -122,9 +122,11 @@ func (p *Provider) loadTLSRoute(listener gatewayListener, route *gatev1alpha2.TL
 			continue
 		}
 
+		rule, priority := hostSNIRule(hostnames)
 		router := dynamic.TCPRouter{
 			RuleSyntax:  "v3",
-			Rule:        hostSNIRule(hostnames),
+			Rule:        rule,
+			Priority:    priority + len(route.Spec.Rules) - ri,
```

Comment:
```suggestion
			Priority:    priority,
```

@augustozanellato The priority calculation for HTTPRoutes produces high values, so using the order of the list as the final grain of salt for differentiation was a workable solution.
Here, the priority is the length of the hostname, if we add the position of the rule to the priority, it is possible to have a route with a wildcard that is prioritised over a full subdomain.